processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -189,7 +189,7 @@ public GroupByQueryQueryToolChest(
                   ImmutableList.<PostAggregator>of(),
                   // Don't do "having" clause until the end of this method.
                   null,
-                  query.getLimitSpec(),
+                  null,
```

Comment:
can we add a comment here, to explain that we don't want to apply the limit until after merging is complete?
This bug only creeped up once we started merging at the historical node level, which caused the limit to be applied before returning results to the broker.